src/core/execute.c

```diff
@@ -739,27 +739,36 @@ static int ask_for_confirmation(const char *vc, const char *cmdline) {
                 goto restore_stdio;
         }
 
-        r = ask_char(&c, "yfs", "Execute %s? [Yes, Fail, Skip] ", e);
-        if (r < 0) {
-                write_confirm_error_fd(r, STDOUT_FILENO);
-                r = 1;
-                goto restore_stdio;
-        }
+        for (;;) {
+                r = ask_char(&c, "yfs?", "Execute %s? [y/f/s/? for help] ", e);
```

Comment:
the uppercase thing is a good point, but I figure the fix for that would be to accept input in both uppercase and lowercase